src/eradius_client.erl

```diff
@@ -313,14 +358,55 @@ code_change(_OldVsn, State, _Extra) -> {ok, State}.
 
 %% @private
 configure(State) ->
-    {ok, ClientPortCount} = application:get_env(eradius, client_ports),
-    {ok, ClientIP} = application:get_env(eradius, client_ip),
-    case parse_ip(ClientIP) of
-        {ok, Address} ->
-            configure_address(State, ClientPortCount, Address);
-        {error, _} ->
-            ?LOG(error, "Invalid RADIUS client IP (parsing failed): ~p", [ClientIP]),
-            {error, {bad_client_ip, ClientIP}}
+    case ets:info(?MODULE) of
+        undefined ->
+            ets:new(?MODULE, [ordered_set, public, named_table, {keypos, 1}, {write_concurrency,true}]),
+            case application:get_env(eradius, servers_pool) of
+                undefined ->
+                    ok;
+                {ok, Pool} ->
+                    lists:foreach(fun(ServersPool) ->
+                        {_PoolName, Servers} = ServersPool,
+                        lists:foreach(fun (Server) ->
+                            case Server of
+                                {Addr, Port, _Secret, Opts} ->
+                                    Retries = proplists:get_value(retries, Opts, ?DEFAULT_RETRIES),
+                                    store_radius_server_from_pool(Addr, Port, Retries);
+                                {Addr, Port, _Secret} ->
+                                    store_radius_server_from_pool(Addr, Port, ?DEFAULT_RETRIES)
+                            end
+                        end, Servers)
+                    end, Pool)
+            end,
```

Comment:
do many levels for one function, break this down to a sensible helper

Reply:
@RoadRunnr yes, you're right. I've reworked this